packages/block-editor/src/components/block-list-appender/index.js

```diff
@@ -50,26 +48,31 @@ function BlockListAppender( {
 		if (
 			! isDocumentAppender &&
 			! isParentSelected &&
-			isAnotherDefaultAppenderAlreadyDisplayed
+			( ! selectedBlockClientId ||
+				isAnotherDefaultAppenderAlreadyDisplayed )
```

Comment:
In case you're wondering, the alteration here prevents non-root-level appenders from rendering when no block is selected.

Reply:
https://github.com/WordPress/gutenberg/pull/25518/files/998456fa95457870be68670707adb056e17cefc9#r492332874